client/sites-dashboard/controller.tsx

```diff
@@ -17,14 +18,20 @@ const globalStyles = css`
 	}
 `;
 
+const getStatusFilterValue = ( status?: string ) => {
+	return (
+		siteStatusFilterValues.find( ( value ) => value === status ) ?? DEFAULT_SITE_STATUS_FILTER_VALUE
+	);
+};
```

Comment:
Hurray for validating the status filter _before_ the page mounts, so we pass a valid value or a fallback to the application and, more importantly, to the components using this information. No need to check down the tree.

Reply:
Nice 😄

Reply:
The downside is that now if we provide the incorrect status, we serve all results regardless of the status. 

From the SEO point of view, a duplicated content issue shouldn't be a problem here, but for the UX, should we redirect to a URL that doesn't have a status parameter in such a case?

Reply:
Could be @wojtekn. I'll try updating this logic to remove the query parameter in case it's `all`.